doc/source/ray-core/user-spawn-processes.rst

```diff
@@ -0,0 +1,106 @@
+
+Both killings happen recursively, meaning if the core worker spawns user process A, then user process A spawns user process B, on core worker exit (or crash) both A and B are killed.
+
+On non-Linux platforms, user-spawned process is not controlled by Ray. The user is responsible for managing the lifetime of the child processes. If the parent Ray worker process dies, the child processes will continue to run.
+
+
+.. contents::
+  :local:
+
+User-Spawned Process Killed on Worker Exit
+------------------------------------------
+
+In the following example, we use Ray Actor to spawn a user process. The user process is a long running process that prints "Hello, world!" every second. The user process is killed when the actor is killed.
+
+.. testcode::
+
+  import ray
+  import psutil
+  import subprocess
+  import time
+  import os
+
+  ray.init(_system_config={"kill_child_processes_on_worker_exit_with_raylet_subreaper":True})
+
+  @ray.remote
+  class MyActor:
+    def __init__(self):
+      pass
+
+    def start(self):
+      # Start a user process
+      process = subprocess.Popen(["/bin/bash", "-c", "sleep 10000"])
+      return process.pid
+
+    def suicide(self):
+      import signal
+      os.kill(os.getpid(), signal.SIGKILL)
+
+
+  actor = MyActor.remote()
+
+  pid = ray.get(actor.start.remote())
+  assert psutil.pid_exists(pid)  # the subprocess running
+
+  actor.suicide.remote()  # sigkill'ed, core worker's subprocess killing no longer works
+  time.sleep(11)  # raylet kills orphans every 10s
+  assert not psutil.pid_exists(pid)
+
+
+Enabling the feature
+-------------------------
+
+To enable the subreaper feature, set the environment variable ``RAY_kill_child_processes_on_worker_exit_with_raylet_subreaper`` to ``true`` **when starting the Ray cluster**, If a Ray cluster is already running, you need to restart the Ray cluster to apply the change. Setting ``env_var`` in a runtime environment will NOT work.
+
+.. code-block:: bash
+
+  RAY_kill_child_processes_on_worker_exit_with_raylet_subreaper=true ray start --head
+
+Another way is to enable it during ``ray.init()`` by adding a ``_system_config`` like this:
+
+.. code-block::
+
+  ray.init(_system_config={"kill_child_processes_on_worker_exit_with_raylet_subreaper":True})
+
+
+⚠️ Caution: Core worker needs to reap zombies
+----------------------------------------------
+
+When the feature is enabled, the core worker process becomes a subreaper (see the next section), meaning there can be some grandchildren processes that are reparented to the core worker process. If these processes exit, core worker needs to reap them to avoid zombies, even though they are not spawn by core worker. If core worker does not reap them, the zombies will accumulate and eventually cause the system to run out of resources like memory.
```

Comment:
We should not mention "core worker". It should be just "worker"

Reply:
renamed all `core worker` to `the worker`